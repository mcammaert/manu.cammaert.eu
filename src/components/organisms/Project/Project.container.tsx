import React from 'react';

import { getProject } from 'api/resume';
import Project from './Project';

import { ProjectContainerProps } from './Project.types';

const ProjectContainer: React.FC<ProjectContainerProps> = ({ project: inputProject, client: inputClient, headerLevel, visible }) => {
  const project = getProject(inputClient, inputProject);
  if (project) {
    const { name, roles, description, references, technologies, tags } = project;
    return (
      <Project
        name={name}
        roles={roles}
        description={description}
        references={references}
        technologies={technologies}
        tags={tags}
        headerLevel={headerLevel}
        visible={visible}
      />
    );
  }
  return <></>;
};

export default ProjectContainer;
