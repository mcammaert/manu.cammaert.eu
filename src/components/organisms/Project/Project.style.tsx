import styled from 'styled-components';
import { modularScale, rem, rgba } from 'polished';

import { Copy } from 'components/atoms/Copy';

export const Project = styled.div`
  border-top: 1px solid ${({ theme }) => rgba(theme.colors.primary, 0.08)};
  margin-top: ${modularScale(0)};
  padding-top: ${modularScale(0)};
  margin-bottom: ${modularScale(0)};
`;

export const Title = styled.div`
  font-size: ${modularScale(1)};
  line-height: ${modularScale(1)};
  font-weight: 200;
  color: ${({ theme }) => theme.colors.secondary};
  margin: 0;
  margin-bottom: ${modularScale(-4)};
`;

export const Roles = styled.ul`
  display: block;
  margin: 0;
  padding: 0;
  margin-bottom: ${modularScale(1)};
`;

export const Role = styled.li`
  font-size: ${modularScale(0)};
  line-height: ${modularScale(1)};
  display: inline-block;

  &:after {
    content: '/';
    padding-left: ${modularScale(-4)};
    padding-right: ${modularScale(-4)};
  }
  &:last-child:after {
    content: '';
    padding 0;
  }
`;

export const Description = styled(Copy)`
  white-space: pre-line;
  margin-top: ${({ theme }) => rem(theme.margin)};
`;

export const Technologies = styled.span`
  padding: 0;
`;

export const Technology = styled.span`
  display: inline-block;
  line-height: ${modularScale(2)};
  margin-right: ${modularScale(-3)};
`;

export const Label = styled.span`
  font-size: ${modularScale(-1)};
  color: ${({ theme }) => theme.colors.secondary};
  background-color: ${({ theme }) => rgba(theme.colors.secondary, 0.05)};
  border-radius: 1em;
  padding: 0.3em 1em;
  font-weight: 400;
  box-shadow: 0 1px 6px ${({ theme }) => rgba(theme.colors.secondary, 0.01)};
  text-transform: uppercase;
`;

export const References = styled.span`
  margin-left: ${({ theme }) => rem(theme.margin)};
  padding: 0;
  margin-bottom: ${modularScale(1)};
  line-height: ${modularScale(3)};
`;

export const Reference = styled.span`
  list-style: none;
  font-size: ${modularScale(0)};
  color: ${({ theme }) => theme.colors.secondary};
  margin-right: ${modularScale(0)};
  border-radius: 5px;
  font-weight: 400;
`;

export default { Project, Title, Roles, Role, Description, Technologies, Technology, Label, References, Reference };
