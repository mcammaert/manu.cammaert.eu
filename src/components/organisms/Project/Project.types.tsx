import { Project } from 'api/resume/resume.types';

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface ProjectProps extends Project {
  headerLevel?: number;
}

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface ProjectContainerProps {
  client: string;
  project: string;
  headerLevel?: number;
}
