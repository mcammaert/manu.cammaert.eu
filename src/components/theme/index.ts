import { createGlobalStyle } from 'styled-components';
import 'normalize.css';
import { colors } from './colors';

export interface GlobalStyleProps {
  readonly light: boolean;
}

// const backgroundColor = (location: string) => (location === '/' ? colors.oxfordBlue : colors.fuzzyWuzzy);

export const GlobalStyle = createGlobalStyle<GlobalStyleProps>`
  body {
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    font-family: 'Nunito Sans', sans-serif;
    width: 100vw;
    background: ${({ light }) => (light ? colors.white : colors.black)};
    background-attachment: fixed;    
    color: ${({ light }) => (light ? colors.black : colors.snow)};
    transition: all 800ms;
    a {
      color: ${({ theme }) => theme.colors.secondary};
    }
  }
`;

export const lightTheme = {
  margin: 15,
  colors: {
    primary: colors.charcoal,
    primaryInvert: colors.snow,
    secondary: colors.primary60,
    secondaryInvert: colors.white,
    tertiary: colors.secondary80,
    tertiaryInvert: colors.white,
    copy: colors.charcoal,
    copyInvert: colors.white,
  },
};

export const darkTheme = {
  margin: 15,
  colors: {
    primary: colors.snow,
    primaryInvert: colors.black,
    secondary: colors.primary30,
    secondaryInvert: colors.primary90,
    tertiary: colors.secondary30,
    tertiaryInvert: colors.secondary80,
    copy: colors.white,
    copyInvert: colors.charcoal,
  },
};
