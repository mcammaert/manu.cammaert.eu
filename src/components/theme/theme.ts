import { createGlobalStyle } from 'styled-components';

export const GlobalStyle = createGlobalStyle`
  body {
    background-color: #ccc;
  }
`;

const theme = {
  main: 'mediumseagreen',
};

export default theme;
