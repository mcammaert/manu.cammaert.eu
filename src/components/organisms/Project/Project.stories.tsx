import React from 'react';

import { boolean, number, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';

import Project from './Project';

storiesOf('Project', module).add('Project', () => {
  return <Project name="Lorem ipsum" />;
});
