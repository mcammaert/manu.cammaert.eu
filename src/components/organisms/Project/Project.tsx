import React from 'react';

import { ScreenReaderOnly } from 'components/atoms/ScreenReaderOnly';

import { ProjectProps } from './Project.types';

import S from './Project.style';

const Project: React.FC<ProjectProps> = ({ name, roles, description, technologies, references }) => (
  <S.Project>
    <S.Title as="h5">{name}</S.Title>
    {roles && (
      <>
        <ScreenReaderOnly>
          <h6>Rollen</h6>
        </ScreenReaderOnly>
        <S.Roles as="ul">
          {roles.map(role => (
            <S.Role key={role} as="li">
              {role}
            </S.Role>
          ))}
        </S.Roles>
      </>
    )}
    {technologies && (
      <>
        <ScreenReaderOnly>
          <h6>TechnologiÃ«n</h6>
        </ScreenReaderOnly>
        <S.Technologies as="ul">
          {technologies.map(technology => (
            <S.Technology key={technology} as="li">
              <S.Label>{technology}</S.Label>
            </S.Technology>
          ))}
        </S.Technologies>
      </>
    )}

    {description && (
      <>
        <ScreenReaderOnly>
          <h6>Omschrijving</h6>
        </ScreenReaderOnly>
        <S.Description>{description}</S.Description>
      </>
    )}
    {references && (
      <>
        <ScreenReaderOnly>
          <h6>Rollen</h6>
        </ScreenReaderOnly>
        <S.References as="ul">
          {references.map(reference => (
            <S.Reference key={reference} as="li">
              <a href={reference}>{reference.replace(/^http(s)?:\/\//i, '')}</a>
            </S.Reference>
          ))}
        </S.References>
      </>
    )}
  </S.Project>
);

export default Project;
