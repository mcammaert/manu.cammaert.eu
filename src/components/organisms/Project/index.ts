import * as S from './Project.style';

export { default as ProjectContainer } from './Project.container';
export { default as Project } from './Project';
export * from './Project.types';
export { S };
