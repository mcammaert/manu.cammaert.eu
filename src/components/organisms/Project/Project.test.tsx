import React from 'react';

import { cleanup } from '@testing-library/react';
import 'jest-styled-components';
import styled from 'styled-components';

import { renderComponent } from 'testing/react-testing-library';
import { lightTheme } from 'components/theme';

import Project from './Project';

afterEach(cleanup);

describe('Project', () => {
  describe('default props', () => {
    it('should match the snapshot', () => {
      const component = <Project name="Lorem ipsum" />;
      const { container } = renderComponent({ theme: lightTheme, component });
      expect(container).toMatchSnapshot();
    });
  });

  describe('all props', () => {
    it('should match the snapshot', () => {
      const component = (
        <Project
          name="Lorem ipsum"
          description="Lorem imsum dolor sit amet"
          technologies={['lorem', 'ipsum', 'dolor']}
          references={['lorem', 'ipsum', 'dolor']}
          roles={['lorem', 'ipsum', 'dolor']}
          headerLevel={3}
          visible
        />
      );
      const { container } = renderComponent({ theme: lightTheme, component });
      expect(container).toMatchSnapshot();
    });
  });

  describe('visibility', () => {
    it('should match the snapshot', () => {
      const component = <Project name="Lorem ipsum" visible={false} />;
      const { container } = renderComponent({ theme: lightTheme, component });
      expect(container).toMatchSnapshot();
    });
  });
});
