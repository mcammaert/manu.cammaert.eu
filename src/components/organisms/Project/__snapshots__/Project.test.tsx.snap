// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project all props should match the snapshot 1`] = `
.c3 {
  position: absolute;
  left: -10000px;
  top: auto;
  width: 1px;
  height: 1px;
  overflow: hidden;
}

.c8 {
  font-size: 0.8661336772616117em;
  line-height: 1.776889em;
  color: #0d5189;
  background-color: rgba(13,81,137,0.1);
  border-radius: 1em;
  padding: 0.25em 1em;
  font-weight: 400;
  box-shadow: 0 1px 6px rgba(13,81,137,0.01);
}

.c1 {
  font-size: 1.333em;
  line-height: 1.333em;
  font-weight: 200;
  color: #262830;
  margin: 0;
  margin-bottom: 0.4221915645190677em;
}

.c9 {
  font-size: 1em;
  font-weight: 200;
  line-height: 1.776889em;
  color: #262830;
  margin-bottom: 0;
}

.c0 {
  background-color: rgba(38,40,48,0.025);
  margin-top: 1em;
  padding: 1em 0.7501875468867217em;
  margin-bottom: 0.7501875468867217em;
  border-radius: 4px;
}

.c2 {
  color: #0d5189;
  margin-bottom: 0;
}

.c4 {
  display: block;
  margin: 0;
  padding: 0;
  margin-bottom: 1.333em;
}

.c5 {
  font-size: 1em;
  line-height: 1.333em;
  display: inline-block;
}

.c5:after {
  content: '/';
  padding-left: 0.3167228541028265em;
  padding-right: 0.3167228541028265em;
}

.c5:last-child:after {
  content: '';
  padding: 0;
}

.c10 {
  white-space: pre-line;
  margin-top: 0.9375rem;
}

.c6 {
  padding: 0;
}

.c7 {
  display: inline-block;
  line-height: 1.776889em;
  margin-right: 0.4221915645190677em;
  margin-bottom: 0.4221915645190677em;
}

.c11 {
  padding: 0;
  line-height: 2.3685930369999997em;
}

.c12 {
  list-style: none;
  font-size: 1em;
  color: #0d5189;
  font-weight: 400;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  width: 90%;
}

<div>
  <div
    class="c0"
  >
    <h3
      class="c1 c2"
      size="1"
    >
      Lorem ipsum
    </h3>
    <div
      class="c3"
    >
      <h4
        class="c1"
        size="1"
      >
        Rollen
      </h4>
    </div>
    <ul
      class="c4"
    >
      <li
        class="c5"
      >
        lorem
      </li>
      <li
        class="c5"
      >
        ipsum
      </li>
      <li
        class="c5"
      >
        dolor
      </li>
    </ul>
    <div
      class="c3"
    >
      <h4
        class="c1"
        size="1"
      >
        TechnologiÃ«n
      </h4>
    </div>
    <ul
      class="c6"
    >
      <li
        class="c7"
      >
        <div
          class="c8"
        >
          lorem
        </div>
      </li>
      <li
        class="c7"
      >
        <div
          class="c8"
        >
          ipsum
        </div>
      </li>
      <li
        class="c7"
      >
        <div
          class="c8"
        >
          dolor
        </div>
      </li>
    </ul>
    <div
      class="c3"
    >
      <h4
        class="c1"
        size="1"
      >
        Omschrijving
      </h4>
    </div>
    <div
      class="c9 c10"
      font-weight="200"
    >
      Lorem imsum dolor sit amet
    </div>
    <div
      class="c3"
    >
      <h4
        class="c1"
        size="1"
      >
        Rollen
      </h4>
    </div>
    <ul
      class="c11"
    >
      <li
        class="c12"
      >
        <a
          href="lorem"
          tabindex="0"
        >
          lorem
        </a>
      </li>
      <li
        class="c12"
      >
        <a
          href="ipsum"
          tabindex="0"
        >
          ipsum
        </a>
      </li>
      <li
        class="c12"
      >
        <a
          href="dolor"
          tabindex="0"
        >
          dolor
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Project default props should match the snapshot 1`] = `
.c1 {
  font-size: 1.333em;
  line-height: 1.333em;
  font-weight: 200;
  color: #262830;
  margin: 0;
  margin-bottom: 0.4221915645190677em;
}

.c0 {
  background-color: rgba(38,40,48,0.025);
  margin-top: 1em;
  padding: 1em 0.7501875468867217em;
  margin-bottom: 0.7501875468867217em;
  border-radius: 4px;
}

.c2 {
  color: #0d5189;
  margin-bottom: 0;
}

<div>
  <div
    class="c0"
  >
    <h2
      class="c1 c2"
      size="1"
    >
      Lorem ipsum
    </h2>
  </div>
</div>
`;

exports[`Project visibility should match the snapshot 1`] = `
.c1 {
  font-size: 1.333em;
  line-height: 1.333em;
  font-weight: 200;
  color: #262830;
  margin: 0;
  margin-bottom: 0.4221915645190677em;
}

.c0 {
  background-color: rgba(38,40,48,0.025);
  margin-top: 1em;
  padding: 1em 0.7501875468867217em;
  margin-bottom: 0.7501875468867217em;
  border-radius: 4px;
}

.c2 {
  color: #0d5189;
  margin-bottom: 0;
}

<div>
  <div
    class="c0"
  >
    <h2
      class="c1 c2"
      size="1"
    >
      Lorem ipsum
    </h2>
  </div>
</div>
`;
